libp2p/Service.cpp

```diff
@@ -127,6 +131,12 @@ void Service::heartBeat()
             SERVICE_LOG(DEBUG) << LOG_DESC("heartBeat ignore invalid address");
             continue;
         }
+        if (m_whitelist != nullptr && !m_whitelist->has(it.second))
+        {
+            SERVICE_LOG(TRACE) << LOG_DESC("heartBeat outside whitelist")
+                               << LOG_KV("nodeid", it.second.abridged());
```

Comment:
If node is not in whitelist, should we disconnect it here?

Reply:
Yes, this logic has been written in line 103
checkWhitelistAndClearSession();